compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -865,9 +865,17 @@ getDefaultDistributedLevelTileSizes(Operation *op,
   SmallVector<int64_t> adjustedMinTileSizes(numLoops, 0);
   SmallVector<int64_t> adjustedMaxTileSizes(numLoops, 0);
   SmallVector<int64_t> adjustedVectorSizeHints(numLoops, 1);
+  // PartitionableLoopsInterface returns partitionable/parallelizable loops from
+  // the innermost to outermost dimensions in sorted order. Get all the
+  // partitionable loops and clamp it for max kNumMaxParallelDims. This
+  // effectively gives kNumMaxParallelDims from outermost to innermost
+  // dimensions.
   SmallVector<unsigned> partitionableLoops =
-      cast<PartitionableLoopsInterface>(op).getPartitionableLoops(
-          kNumMaxParallelDims);
+      cast<PartitionableLoopsInterface>(op).getPartitionableLoops(std::nullopt);
```

Comment:
One point to consider, when I initially wrote it, the idea was that for the parallel loops that are not partitionable you would set the tile size to 1. With that in mind it is better to take the inner parallel loops and distribute the outer parallel. I am fine either way though.

Reply:
This patch distributes the outer parallel dim. For an op such as attention, where the parallel dimension is> 3, it's better to distribute the batch dimension, i.e., the 0th dim. Or, as @hanhanW mentioned, we can limit this case to the Attention op.